tests/Integration/Behaviour/Features/Context/Domain/Product/DuplicateProductFeatureContext.php

```diff
@@ -0,0 +1,51 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace Tests\Integration\Behaviour\Features\Context\Domain\Product;
+
+use PrestaShop\PrestaShop\Core\Domain\Product\Command\DuplicateProductCommand;
+
+class DuplicateProductFeatureContext extends AbstractProductFeatureContext
+{
+    /**
+     * @When I duplicate product :productReference to a :newProductReference
+     */
+    public function duplicate(string $productReference, string $newProductReference): void
+    {
+        $newProductId = $this->getCommandBus()->handle(new DuplicateProductCommand(
+            $this->getSharedStorage()->get($productReference)
+        ));
+
+        $this->getSharedStorage()->set($newProductReference, $newProductId->getValue());
+    }
+
+    public function assertDuplicatedProduct(): void
+    {
+        //@todo: how do I assert duplicated product props? The Issue - product properties are dissolved & asserted each in separated contexts
```

Comment:
> how do I assert duplicated product props?

What about we have this `assertDuplicatedProduct()` and its logic does rely on other contexts ?
```php
public function assertDuplicatedProduct($oldProductId, $newProductId): void
{
    $anotherContext->assertDescription(...);
    $anotherContext->...
    ...
}
```

Well, according to https://github.com/Behat/Behat/issues/637 it's actually a design flaw 🤔  but this issue suggests something else: start creating a ProductAssertContextRepository class where we store all product asserts and every Context use it.

"Much better option is to treat your context files as controllers and delegate all the work to other layers. You don't need to put everything in your context files, instead you can create other objects and inject them into your contexts. It's very easy with the extension system behat provides (only drawback is it's not very well documented yet)."